From 0c6f97aee2005e64398449b3e326afd99630fd4b Mon Sep 17 00:00:00 2001
From: Luca Stefani <luca.stefani.ge1@gmail.com>
Date: Fri, 19 Oct 2018 12:22:28 +0200
Subject: [PATCH 3/4] Merge android-9.0.0_r12

commit 83e2a83dd41ae9c7376e7735a6345dae1c71406a
Merge: 932eb00d b8d86c00
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Wed Jul 4 03:01:20 2018 +0000

    Snap for 4876468 from b8d86c0051a0113659edf2bf7ea80b6d318e01f5 to pi-dr1-release

    Change-Id: I262a45bdb491e2333aae553cadaa6048ae86c41d

commit b8d86c0051a0113659edf2bf7ea80b6d318e01f5
Author: Yabin Cui <yabinc@google.com>
Date:   Tue May 1 17:31:36 2018 -0700

    simpleperf: fix test on APCT.

    Bug: http://b/77470883
    Test: run adb shell
    Test: /data/local/tmp/simpleperf_unit_test/simpleperf_unit_test.
    Change-Id: I640ad76b68842045f6ecdcb3cf686b7823d7d5db
    (cherry picked from commit 03a31c13168f6502883e8beb4dbc76bc064a54a0)

Change-Id: I4226c1b90969d6124800e29b4ecd071a619860bb
---
 simpleperf/cmd_record_test.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/simpleperf/cmd_record_test.cpp b/simpleperf/cmd_record_test.cpp
index 7f208862..31f17278 100644
--- a/simpleperf/cmd_record_test.cpp
+++ b/simpleperf/cmd_record_test.cpp
@@ -270,7 +270,10 @@ TEST(record_cmd, existing_threads) {
   ASSERT_TRUE(RunRecordCmd({"-t", tid_list}));
 }
 
-TEST(record_cmd, no_monitored_threads) { ASSERT_FALSE(RecordCmd()->Run({""})); }
+TEST(record_cmd, no_monitored_threads) {
+  TemporaryFile tmpfile;
+  ASSERT_FALSE(RecordCmd()->Run({"-o", tmpfile.path}));
+}
 
 TEST(record_cmd, more_than_one_event_types) {
   ASSERT_TRUE(RunRecordCmd({"-e", "cpu-cycles,cpu-clock"}));
-- 
2.17.1

